### PR TITLE
inih: init at r52

### PR DESCRIPTION
**Pull request by TredwellGit**
###### Motivation for this change
Needed for https://github.com/NixOS/nixpkgs/issues/84682 and to update xfsprogs.

###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by TredwellGit** [comment]
Sorry for the base change.

File: pkgs/development/libraries/inih/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "-Ddefault_library=shared"
+    "-Ddistro_install=true"
+    "-Dwith_INIReader=true"
+  ];
+
+  meta = with stdenv.lib; {
+    description = "Simple .INI file parser in C, good for embedded systems";
+    homepage = "https://github.com/benhoyt/inih";
+    changelog = "https://github.com/benhoyt/inih/releases/tag/${version}";
+    license = licenses.bsd3;
```
Comment (Mic92) [inline pkgs/development/libraries/inih/default.nix:26]: Are you going to maintain this package?

**Comment by TredwellGit** [comment]
Force pushed to make stuck ofborg try again.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107010` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>inih</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107010` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>inih</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
> Force pushed to make stuck ofborg try again.

you can also do
```
@ofborg eval
```

File: pkgs/development/libraries/inih/default.nix (hunk 1 above)
Comment (TredwellGit) [reply pkgs/development/libraries/inih/default.nix:26]: Let me add myself as a maintainer.

**Comment by lovesegfault** [comment]
Result of `nixpkgs-review pr 107010` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>inih</li>
  </ul>
</details>

**Pull request merged**